src/main/products/coin/ClientCoinArrayList.java

```diff
@@ -0,0 +1,114 @@
+package main.products.coin;
+
+import lombok.Getter;
+import main.enums.CoinType;
+
+/**
+ * A custom collection implementation for storing Coin objects.
+ * Provides methods to put, to get, delete coins
+ * and exchange coins for bonus when 10 coins of the same type are accumulated.
+ * Handles dynamic resizing with an increment of +10.
+ */
+@Getter
+public class ClientCoinArrayList {
```

Comment:
Upu can try to do it generic. Now we can use it only with coins

Reply:
If it were generic, I think I would lose all my coin exchange logic.

Reply:
Ok) it's your decision to make it with coins) let's keep it that way) it's ok for me too)